### PR TITLE
Feature/improved search

### PR DESCRIPTION
**Pull request by palmenhq**

### CODE REVIEW

**Comment by FredrikAreschoug** [comment]
Looks good to me :)

File: member-service/src/main/java/com/hedvig/memberservice/query/MemberRepository.kt [hunk 1]
```diff
@@ -42,15 +42,58 @@ interface MemberRepository : JpaRepository<MemberEntity, Long> {
...
+                CAST(m.id as text) = :query
+                OR lower(m.firstName) LIKE ('%' || lower(:query) || '%')
+                OR lower(m.lastName) LIKE ('%' || lower(:query) || '%')
+                OR lower(m.firstName || ' ' || m.lastName) LIKE ('%' || lower(:query) || '%')
+                OR m.ssn LIKE ('%' || :query || '%')
+                OR lower(m.email) LIKE ('%' || lower(:query) || '%')
+                OR m.phoneNumber LIKE ('%' || :query || '%')
+            )
+    """)
+    fun searchSignedOrTerminatedByIdAndSsnAndNameAndEmail(
```
Comment (meletios) [inline member-service/src/main/java/com/hedvig/memberservice/query/MemberRepository.kt]:
```suggestion
    fun searchSignedOrTerminatedByIdOrSsnOrNameOrEmail(
```

Comment (meletios) [reply member-service/src/main/java/com/hedvig/memberservice/query/MemberRepository.kt]: Shouldnt it be more accurate?

File: member-service/src/main/java/com/hedvig/memberservice/query/MemberRepository.kt [hunk 2]
```diff
@@ -42,15 +42,58 @@ interface MemberRepository : JpaRepository<MemberEntity, Long> {
...
+        WHERE
+            CAST(m.id as text) = :query
+            OR lower(m.firstName) LIKE ('%' || lower(:query) || '%')
+            OR lower(m.lastName) LIKE ('%' || lower(:query) || '%')
+            OR lower(m.firstName || ' ' || m.lastName) LIKE ('%' || lower(:query) || '%')
+            OR m.ssn LIKE ('%' || :query || '%')
+            OR lower(m.email) LIKE ('%' || lower(:query) || '%')
+            OR m.phoneNumber LIKE ('%' || :query || '%')
+    """)
+    fun searchAllByIdAndSsnAndNameAndEmail(
```
Comment (meletios) [inline member-service/src/main/java/com/hedvig/memberservice/query/MemberRepository.kt]:
```suggestion
    fun searchAllByIdOrSsnOrNameOrEmailOrPhoneNumber(
```

File: member-service/src/main/java/com/hedvig/memberservice/services/member/MemberQueryService.java [hunk 3]
```diff
@@ -48,43 +48,19 @@ public InternalMemberSearchResultDTO search(InternalMemberSearchRequestDTO searc
 
     if (request.getSortBy() != null) {
       String entSortProp = MemberEntity.SORT_COLUMN_MAPPING.get(request.getSortBy());
-      pageableBuilder = pageableBuilder.orderBy(entSortProp, request.getSortDirection(), Sort.NullHandling.NULLS_LAST);
+      pageableBuilder = pageableBuilder.orderBy(entSortProp, request.getSortDirection(), Sort.NullHandling.NULLS_FIRST);
```
Comment (meletios) [inline member-service/src/main/java/com/hedvig/memberservice/services/member/MemberQueryService.java]: Why first?

Comment (palmenhq) [reply member-service/src/main/java/com/hedvig/memberservice/services/member/MemberQueryService.java]: We're defaulting to `desc` in the sort direction and we want the nulls _really_ last haha

**Pull request merged**